ingest.py

```diff
@@ -65,7 +68,11 @@ def load_documents(source_dir: str) -> List[Document]:
         all_files.extend(
             glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
         )
-    return [load_single_document(file_path) for file_path in all_files]
+    pool = multiprocessing.Pool(os.environ.get('LOAD_DOCUMENTS_NUMBER_OF_THREADS', multiprocessing.cpu_count()))
+    result = pool.map(load_single_document, all_files)
+    pool.close()
+    pool.join()
```

Comment:
It is also typical to use `with` for anything like this for exception handling. example code would look like this:
```
with multiprocessing.Pool(os.environ.get('LOAD_DOCUMENTS_NUMBER_OF_THREADS', multiprocessing.cpu_count()-1)) as pool:
    pool.map(load_single_document, all_files)
```
besides that everything else looks good

Reply:
Thanks, I did a commit fixing it